src/libraries/System.IO.FileSystem/tests/File/GetSetTimes.cs

```diff
@@ -32,129 +32,140 @@ protected override string GetExistingItem(bool readOnly = false)
 
         protected override string CreateSymlink(string path, string pathToTarget) => File.CreateSymbolicLink(path, pathToTarget).FullName;
 
-        [Fact]
-        [PlatformSpecific(TestPlatforms.Linux)]
-        public void BirthTimeIsNotNewerThanLowestOfAccessModifiedTimes()
-        {
-            // On Linux, we synthesize CreationTime from the oldest of status changed time and write time
-            //  if birth time is not available. So WriteTime should never be earlier.
+        protected abstract void SetCreationTime(string path, DateTime creationTime);
 
-            // Set different values for all three
-            // Status changed time will be when the file was first created, in this case)
-            string path = GetExistingItem();
-            File.SetLastWriteTime(path, DateTime.Now.AddMinutes(1));
-            File.SetLastAccessTime(path, DateTime.Now.AddMinutes(2));
+        protected abstract DateTime GetCreationTime(string path);
 
-            // Assert.InRange is inclusive.
-            Assert.InRange(File.GetCreationTimeUtc(path), DateTime.MinValue, File.GetLastWriteTimeUtc(path));
-        }
+        protected abstract void SetCreationTimeUtc(string path, DateTime creationTime);
 
-        [Fact]
-        [PlatformSpecific(TestPlatforms.Linux)]
-        public async Task CreationTimeSet_GetReturnsExpected_WhenNotInFuture()
-        {
-            // On Linux, we synthesize CreationTime from the oldest of status changed time (ctime) and write time (mtime).
-            // Changing the CreationTime, updates mtime and causes ctime to change to the current time.
-            // When setting CreationTime to a value that isn't in the future, getting the CreationTime should return the same value.
+        protected abstract DateTime GetCreationTimeUtc(string path);
 
-            string path = GetTestFilePath();
-            File.WriteAllText(path, "");
+        protected abstract void SetLastAccessTime(string path, DateTime creationTime);
```

Comment:
The `DateTime` argument names don't match with the functions. `creationtime` should be `lastAccessTime`. Or maybe you want to changes them all to `dateTime`.